Dockerfile

```diff
@@ -198,7 +163,40 @@ RUN python3 -m pip install *.whl && rm *.whl
 #
 # Make sure the ash script is executable
 #
-RUN chmod -R +r /ash && chmod +x /ash/ash
+RUN chmod -R 777 /ash /src /out /deps && chmod +x /ash/ash
```

Comment:
Should we separate this? I don't think we need execution permissions on `/src` or `/out` (deps and ash folder probably yes.

Reply:
755 (or 750) should be enough for `/ash`, `/src` .  If ash writes to `/deps`, then it will need 777 (or at least 770 if the UID/GID is "right").

Reply:
@climbertjh2 @rafaelpereyra - Part of the problem with this is the CI compatibility part. We don't 100% know the users that will run this container, but we do know that it does require explicit permissions to at least read/write to these paths. I think we can likely do 755 for everything and grant 777 where needed within /ash. Some areas where write access is needed for Others include dependencies that are pulled at runtime (Grype DB when not in offline mode, Semgrep rules, etc). `770` would be nice, but at this stage in the image build we do not know exactly which user will end up interacting within the container. We could further restrict these dirs in the final stage as well to 750/770, as at _that_ stage we do indeed know which user is running the container.

Reply:
Ok, that makes sense to me.

Reply:
![CleanShot 2024-10-31 at 21 27 39](https://github.com/user-attachments/assets/ab03ca5f-f74e-48a7-9908-2e658c3eff35)

@rafaelpereyra @climbertjh2 - Let me know if this is good to resolve now!

Reply:
I'm ok with the explanation too. The UID/GID uncertainty is definitively a challenge.